### PR TITLE
Moved to hyper 0.11 and serde 1.0

### PR DESCRIPTION
**Pull request by GyrosOfWar**
Hi! I updated the library versions of serde and hyper. That means the `build.rs` script is no longer necessary and we need a dependency on a HTTPS connector (in this case I chose hyper-native-tls). I also added some custom code to deserialize reddit's timestamps (since for some reason, they sometimes have a trailing `.0`)

### CODE REVIEW

**Comment by jackinloadup** [comment]
@GyrosOfWar note that this doesn't update to hyper 0.11. Actually up to  0.10.

**Comment by jackinloadup** [comment]
@GyrosOfWar note that this doesn't update to hyper 0.11. Actually up to  0.10.

**Comment by GyrosOfWar** [comment]
Whoops, you're right.

File: Cargo.toml [hunk 1]
```diff
@@ -6,18 +6,16 @@ license = "MIT"
...
 
-[build-dependencies]
-serde_codegen = "0.8"
-
 [dependencies]
-hyper = "0.9.10"
-serde = "0.7.14"
-serde_json = "0.7.4"
+hyper = "0.10"
+hyper-native-tls = "0.2"
```
Comment (johan-bjareholt) [inline Cargo.toml:18]: Doesn't compile on machines with OpenSSL 1.1+ (same as rawr without this PR), simply changing it to "0.3" worked very well for me!

**Review by johan-bjareholt — COMMENTED** [review]
Using this patch and bumping hyper-native-tls to 0.3 fixes compilation issues for me on master!

Would love to have this merged!

**Pull request closed without merging**